modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/clip/ShareClipViewModel.kt

```diff
@@ -98,6 +124,109 @@ class ShareClipViewModel @AssistedInject constructor(
         clipPlayer.setPlaybackPollingPeriod(pollingPeriod)
     }
 
+    fun shareClip(
+        podcast: Podcast,
+        episode: PodcastEpisode,
+        clipRange: Clip.Range,
+        platform: SocialPlatform,
+        cardType: CardType,
+        sourceView: SourceView,
+        createBackgroundAsset: suspend (VisualCardType) -> Result<File>,
+    ) {
+        if (isSharingState.value) {
+            return
+        }
+        isSharingState.value = true
+        viewModelScope.launch {
+            val animation = async {
+                // Make sure that animation lasts for at least 1 seconds when sharing
+                // something other than a link
+                if (platform != SocialPlatform.PocketCasts) {
+                    delay(1.seconds)
+                }
```

Comment:
Why do we need to wait a second? Does it look strange if it flashes on the screen too quickly?

Reply:
Yeah, it can look ugly if it just flashes between two states.